test: Added AOI random kernel

TwoTenPvP opened the pull request:
Adds an InterestKernel that randomly decides which objects should be visible on every query.

TwoTenPvP committed:
test: Added AOI random kernel

mattwalsh-unity commented on com.unity.multiplayer.mlapi/Tests/Runtime/AOI/RandomInterestKernel.cs:
```diff
@@ -0,0 +1,25 @@
+using MLAPI.Interest;
+using UnityEngine;
+
+namespace MLAPI.RuntimeTests.AOI
```
Do you think we should have a namespace just for AOI?  If so it would seem you'd also want to change `InterestTests.cs` below to be in that namespace?

TwoTenPvP replied:
Perhaps, I tried to not touch the existing stuff.

mattwalsh-unity replied:
Ok, I'd like to either make your Kernel file just be `MLAPI.RuntimeTests` or fix `InterestTests.cs` to be `MLAPI.RuntimeTests.AOI`.  

And actually in the latter case, it would be `MLAPI.RuntimeTests.Interest`.  I moved from AOI->Interest

JesseOlmer replied:
Moved the tests from "AOI" to "Interest" directory and changed namespace to "MLAPI.RuntimeTests". We can deal with the "Interest" namespace in a separate PR

mattwalsh-unity commented on com.unity.multiplayer.mlapi/Tests/Runtime/AOI/RandomInterestKernel.cs:
```diff
@@ -0,0 +1,25 @@
+
+        public override void QueryFor(in NetworkClient client, in NetworkObject obj, HashSet<NetworkObject> results)
+        {
+            VisibleObjects.Clear();
```
I assume VisibleObjects is just used for testing.  I was about to say, is there a clever way to have VisibleObjects only in play for testing?

JesseOlmer replied:
Removed this type as it didn't work right before, and it's no longer needed in the new test suite

mattwalsh-unity commented on com.unity.multiplayer.mlapi/Tests/Runtime/InterestTests.cs:
```diff
@@ -129,6 +130,58 @@ public override void UpdateObject(in NetworkObject obj)
+            results.Clear();
+            NetworkManager.Singleton.InterestManager.QueryFor(nc, results);
+
+            Assert.True(!randomKernel.VisibleObjects.Contains(random1Obj) || results.Contains(random1Obj));
```
Ok, if I read this right, this makes sure that the kernel gets invoked and therefore VisibleObjects needs to match the results. 

But if we're trying to test the random aspect, this could be stronger.  What I would probably do is:

1. Add a "set seed" method to the RandomInterestKernel
2. Then in your test, call Random.Value n times (where n is substantially more than 3).  Now you know how many to expect to come through
3. Now call setSeed again, and run the tests, and make sure you get the same number of hits
4. I think with this you can also eliminate the need for the VisibleObjects tracker

JesseOlmer replied:
Took a slightly different route on the test coverage, but hopefully end up with the same or better functional validation

TwoTenPvP committed:
Improved test

Commit:
test: Rewrite RandomInterestKernel test suite

- monolithic integration test removed
- several unit tests added
- tests moved from runtime (play mode) to editor (edit mode) assemblies

Commit:
test: Remove VisibleObjects from RandomInterestKernel

Removed the list as it never worked "right", and with the new test suite it's not needed at all.
